core/kernel/tee_misc.c

```diff
@@ -94,7 +94,7 @@ static bool is_valid_conf_and_notnull_size(
 		vaddr_t b, size_t bl, vaddr_t a, size_t al)
 {
 	/* invalid config return false */
-	if ((b + bl < b) || (a + al < a))
+	if ((b - 1 + bl < b) || (a - 1 + al < a))
```

Comment:
Is this really correct? I believe the first check here is to find out if we wrap around the end of memory.

To make the numbers easy let's say our memory is only 8 bytes big (instead of 32-bit). So when addresses are 0-7.

```
b  = 4
bl = 4

Original code:
if (b + bl < b) ...
[0|1|2|3|4|5|6|7]
 |       |       |
 |       b       b+bl
 |
 b+bl <-- wrap detected, false is returned

Updated code:
if (b - 1 + bl < b)
[0|1|2|3|4|5|6|7]
         |     |
         b     b-1+bl <-- No wrap detected, meaning that this will return true?
```
This isn't correct or am I wrong?